simple_parsing/wrappers/dataclass_wrapper.py

```diff
@@ -77,7 +78,13 @@ def __init__(
                 child_wrapper.required = False
                 child_wrapper.optional = True
                 self._children.append(child_wrapper)
-
+            elif utils.is_final(field.type):
+                # Final fields have no business being set by the command line
+                continue
```

Comment:
I don't think this is dataclass way.
Currently dataclasses do not use `Final` at all.
`Final` means variable should not be changed after initialization.
But it must be initialized so it's not the same as skippable `field(init=False)` or `ClassVar`

Reply:
Final fields are being used in classes and dataclasses to express a constant field/member like in typesafe languages such as Java, so I think it makes sense for SP not to do the wrong thing. A Final field must be initialized with a value at the point of declaration, of course, however, the point of the PR was to support a literal as a shorter and more standardized idiom in addition to choice().

If you think SP shouldn't support Finals at all, that's fine by me, but it would do the wrong thing in case a client *does* choose to use Final.

Reply:
Let me see if I understand the options here (and please correct me if I'm wrong):

Option 1: (@idoby) Don't create command-line options for Final fields. (Since they are normally used as constant attributes on dataclasses)

Option 2: (@mixilchenko) Create command-line options for Final fields, since the Final annotation means "constant after initialization".

Did I understand that correctly?

Reply:
On second thought, I'd prefer option 2 because option 1 views the dataclasses as specific to the command line, which I would like to avoid (I see them as domain-driven objects that SP lets me conveniently pull from the command line), but this is confusing so we'll need to document this very very well.

Reply:
Also, here's one good thing about option 2: creating the arguments for a field like  `a: Final[SomeType]` will be exactly the same as creating the arguments for a `a: SomeType` field!

Reply:
Well, you'd still need to extract the actual type out of the Final but yeah it's a very simple change